torchvision/utils.py

```diff
@@ -179,12 +183,21 @@ def draw_bounding_boxes(
 
     img_boxes = boxes.to(torch.int64).tolist()
 
-    draw = ImageDraw.Draw(img_to_draw)
+    if fill:
+        draw = ImageDraw.Draw(img_to_draw, "RGBA")
+
+    else:
+        draw = ImageDraw.Draw(img_to_draw)
+
     txt_font = ImageFont.load_default() if font is None else ImageFont.truetype(font=font, size=font_size)
 
     for i, bbox in enumerate(img_boxes):
         color = None if colors is None else colors[i]
-        draw.rectangle(bbox, width=width, outline=color)
+        if fill:
+            fill_color = color + [100]
```

Comment:
I think we would need to set alpha channel to some hardcoded value. Since that gives the semi transparent look.

Reply:
I just noticed that this causes issues on the [type checks](https://app.circleci.com/pipelines/github/pytorch/vision/5971/workflows/02e568f7-273c-4816-ba89-5dbe0a001d01/jobs/385457). Also this is not going to work for all values of `color`. There are various corner cases:
1. The `color` could be None
2. It could be a string
3. Or a tuple.

It might be worth confirming that all work as expected by adding a test, what you think?

Reply:
If fill is true

If color is `None` lets fill with default color. 

If color is `Tuple[int, int, int]` I think we just add alpha of 100 as above and do. 

If color is `str` I think we need to get RGB value of the str add default alpha of 100 and draw ?

Reply:
I think we need to handle the `fill_color = color + [100]` especially in each corner case:

```
>>> None + [100]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: unsupported operand type(s) for +: 'NoneType' and 'list'

>>> (0, 0, 0) + [100]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: can only concatenate tuple (not "list") to tuple

>>> "red" + [100]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: can only concatenate str (not "list") to str
```

Reply:
Sure. Sorry, I'm bit low on bandwidth this week but will complete by next week.
I will add tests too to check these corner cases.

Reply:
Don't worry mate. :) Take your time! Ping me when ready to make sure I review it again. Thanks!